cmd/globals.go

```diff
@@ -200,5 +203,17 @@ func setGlobalsFromContext(ctx *cli.Context) error {
 			globalResolvers[host] = addr
 		}
 	}
+
+	if ctx.IsSet("header") {
```

Comment:
```suggestion
	if ctx.IsSet("header") {
```

You don't need this, globalCustomHeader will be empty if there are no `StringSlice` entries, I would also suggest to avoid using a global value instead pass it as part of the `type S3Client struct` and pass it around this way.. as it would be safer.